Fix 404 Page not displaying

MrDirkelz opened the pull request.

MrDirkelz committed:
Remove unnecessary redirects in SingleContent.vue to allow 404 logic to handle missing content

MrDirkelz committed:
Add test to verify 404 error page displays for invalid slug without redirecting to homepage

ivanslabbert commented on app/src/pages/SingleContent.spec.ts:
Add a test that will ensure that a slug renders correctly when not in IndexedDb but available in the online API

ivanslabbert reviewed (commented):
Momentarily shows 404 error before loading content from slug. We need to wait until the API confirmed that the slug is not available before showing an 404. When offline however, we show the 404 if the slug is not in IndexedDb

MrDirkelz committed:
feat: implement SingleContent page with content loading and redirection logic

- Added SingleContent.vue component for displaying content details.
- Implemented useContentLoader helper for fetching content from IndexedDB and API.
- Created unit tests for useContentLoader to ensure correct behavior in various scenarios.
- Updated router to point to the new SingleContent.vue file.

MrDirkelz committed:
Unskip notification test for preferred language

Unskip the test for showing notification in preferred language.

MrDirkelz committed:
feat: enhance SingleContent component with dynamic translation loading and language metadata retrieval

DirkDevelopmentFreelance commented:
This was too complex of a solution
